pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
This is not good: if the error is "you're outside the watch window" you will just retry forever.